src/collectors/snmpinterfacepoll/snmpinterfacepoll.py

```diff
@@ -73,11 +73,15 @@ def publish_metrics(self, gaugesList, countersList, timestamp):
 
         for gaugeDesc in gaugesList:
             metricPath, metricValue = gaugeDesc
-            metric = Metric(
-                            metricPath, metricValue,
-                            raw_value=None, timestamp=timestamp
-                            )
-            self.publish_metric(metric)
+            # Fixes devices that return no value
+            try:
+                metric = Metric(
+                                metricPath, metricValue,
+                                raw_value=None, timestamp=timestamp
+                                )
+                self.publish_metric(metric)
+            except AttributeError:
```

Comment:
If the network is down then none of your metrics are going to get published either way. But I see what you're saying this can be a generic exception catching if you like to be, in my case it wasn't.